From 3a5929942f89b48136583e94c4c8e593b532f16c Mon Sep 17 00:00:00 2001
From: Demi Marie Obenour <demi@invisiblethingslab.com>
Date: Fri, 2 Apr 2021 22:13:15 -0400
Subject: [PATCH 12/22] Require package headers to be more than a region

A package header that is only a region might need special handling
elsewhere, and will never happen in practice.  Just reject it.
---
 lib/header.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/header.c b/lib/header.c
index d1d37ff..9f47b19 100644
--- a/lib/header.c
+++ b/lib/header.c
@@ -1881,7 +1881,8 @@ static rpmRC hdrblobVerifyRegion(rpmTagVal regionTag, int exact_size,
     }
 
     /* In package files region size is expected to match header size. */
-    if (exact_size && !(blob->il == blob->ril && blob->dl == blob->rdl)) {
+    if (exact_size &&
+	!(blob->il == blob->ril && blob->dl == blob->rdl && blob->il > 1)) {
 	rasprintf(buf,
 		_("region %d: tag number mismatch il %d ril %d dl %d rdl %d\n"),
 		regionTag, blob->il, blob->ril, blob->dl, blob->rdl);
-- 
2.30.2

